storage/memory/log_storage.go

```diff
@@ -342,6 +342,22 @@ func (t *logTreeTX) GetActiveLogIDs(ctx context.Context) ([]int64, error) {
 	return t.getActiveLogIDs(ctx)
 }
 
+func (t *readOnlyLogTX) GetUnsequencedCounts(ctx context.Context) (map[int64]int64, error) {
+	t.ms.mu.RLock()
+	defer t.ms.mu.RUnlock()
+
+	ret := make(map[int64]int64)
+	for id, tree := range t.ms.trees {
+		tree.mu.RLock()
+		defer tree.mu.RUnlock()
```

Comment:
Yup, I know defers are func-level, this just locks all the trees it touches until it returns when they all become unlocked - I figured that was fine for a test impl, and led to fewer lines of code.